### DISCUSSION TITLE
❓ [QUESTION] About parity in irreducible representation

### DISCUSSION THREAD

**Discussion by hyuntae-cho** [mir-group/nequip#246] [Q&A] [answered]
For O(3) representation, parity variable is adopted which represents reflections.
From what I've understand, when we apply (x,y,z) |-> (-x,-y,-z) action to all coordinates, every parity of feature representations should change. 
However, it seems `nequip.nn.embedding.SphericalHarmonicEdgeAttrs` always calculate edge attribute with fixed irreducible representations. 
Could you explain more about it? Thanks for your time! :)

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi @choant ,

This is almost correct--- the key is that the parity in the _irrep_ does not describe the sign of the data itself, but rather how that sign should transform under inversions. The L=1 spherical harmonic, for example, is a vector and thus has odd parity (`-1`); this means that when you invert the input coordinates the sign of the L=1 tensor flips. What parity `-1` in the irrep signifies is that this flip occurs, not the current "state" of the sign for some particular data that transforms according to the irrep.
